receiver/otelarrowreceiver/internal/logs/otlp.go

```diff
@@ -37,7 +44,17 @@ func (r *Receiver) Export(ctx context.Context, req plogotlp.ExportRequest) (plog
 	}
 
 	ctx = r.obsrecv.StartLogsOp(ctx)
-	err := r.nextConsumer.ConsumeLogs(ctx, ld)
+
+	sizeBytes := int64(r.sizer.LogsSize(req.Logs()))
+	err := r.boundedQueue.Acquire(ctx, sizeBytes)
+	if err != nil {
+		return plogotlp.NewExportResponse(), err
+	}
+	defer func() {
+		err = r.boundedQueue.Release(sizeBytes)
```

Comment:
I think it would be better not to modify the `err` here, so the caller gets the status of their operation. The only failures inside boundedQueue.Release() are internal bugs, so logging them as errors would be appropriate.

Reply:
Fixed